### PR TITLE
Fix build with nginx-1.9.11+.

### LINKED ISSUE #1110: Fix building with nginx 1.9.11+ and support dynamic module loading
Hi, I'm using the latest changeset from Nginx's [repository](https://trac.nginx.org/nginx/log/nginx/) and building it from source with the latest ngx_pagespeed and headers-more-nginx. Whilst compiling I get the error:

```
cc -c -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g -g -O2  -ffunction-sections -fdata-sections -fstack-protector-strong --param=ssp-buffer-size=4 -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2  -D_GLIBCXX_USE_CXX11_ABI=0  -I src/core -I src/event -I src/event/modules -I src/os/unix -I ../ngx_pagespeed-release-1.10.33.4-beta/psol/include -I ../ngx_pagespeed-release-1.10.33.4-beta/psol/include/third_party/chromium/src -I ../ngx_pagespeed-release-1.10.33.4-beta/psol/include/third_party/google-sparsehash/src -I ../ngx_pagespeed-release-1.10.33.4-beta/psol/include/third_party/google-sparsehash/gen/arch/linux/x64/include -I ../ngx_pagespeed-release-1.10.33.4-beta/psol/include/third_party/protobuf/src -I ../ngx_pagespeed-release-1.10.33.4-beta/psol/include/third_party/re2/src -I ../ngx_pagespeed-release-1.10.33.4-beta/psol/include/out/Debug/obj/gen -I ../ngx_pagespeed-release-1.10.33.4-beta/psol/include/out/Debug/obj/gen/protoc_out/instaweb -I ../ngx_pagespeed-release-1.10.33.4-beta/psol/include/third_party/apr/src/include -I ../ngx_pagespeed-release-1.10.33.4-beta/psol/include/third_party/aprutil/src/include -I ../ngx_pagespeed-release-1.10.33.4-beta/psol/include/third_party/apr/gen/arch/linux/x64/include -I ../ngx_pagespeed-release-1.10.33.4-beta/psol/include/third_party/aprutil/gen/arch/linux/x64/include -I objs -I src/http -I src/http/modules -I src/http/v2 \
        -o objs/addon/src/ngx_pagespeed.o \
        ../ngx_pagespeed-release-1.10.33.4-beta/src/ngx_pagespeed.cc

../ngx_pagespeed-release-1.10.33.4-beta/src/ngx_pagespeed.cc:3133:1: error: deprecated conversion from string constant to ‘char*’ [-Werror=write-strings]
 };
 ^
../ngx_pagespeed-release-1.10.33.4-beta/src/ngx_pagespeed.cc:3148:1: error: deprecated conversion from string constant to ‘char*’ [-Werror=write-strings]
 };
 ^
```

my configure: 

```
./configure --prefix=/etc/nginx --sbin-path=/usr/sbin/nginx --conf-path=/etc/nginx/nginx.conf --error-log-path=/var/log/nginx/error.log --http-log-path=/var/log/nginx/access.log --pid-path=/var/run/nginx.pid --lock-path=/var/run/nginx.lock --http-client-body-temp-path=/var/lib/nginx/body --http-proxy-temp-path=/var/lib/nginx/proxy --http-fastcgi-temp-path=/var/lib/nginx/fastcgi --user=nobody --group=nobody --with-http_geoip_module --with-http_ssl_module --with-http_v2_module  --with-http_realip_module --with-http_gunzip_module --with-http_gzip_static_module --with-http_stub_status_module --with-ipv6 --with-threads --with-file-aio --with-libatomic --with-debug --with-pcre --with-pcre-jit --without-http_empty_gif_module --without-http_memcached_module --without-http_scgi_module --without-http_split_clients_module --without-http_ssi_module --without-http_upstream_hash_module --without-http_upstream_ip_hash_module --without-http_upstream_least_conn_module --without-http_userid_module --without-http_uwsgi_module --add-module=../ngx_pagespeed-release-${NPS_VERSION}-beta  --add-module=../headers-more-nginx-module --with-cc-opt='-g -O2  -ffunction-sections -fdata-sections -fstack-protector-strong --param=ssp-buffer-size=4 -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2' --with-ld-opt="-Wl,--gc-sections,-s,-z,relro,--as-needed"
```

I also noticed during the configuration this:
```
adding module in ../ngx_pagespeed-release-1.10.33.4-beta
mod_pagespeed_dir=../ngx_pagespeed-release-1.10.33.4-beta/psol/include
build_from_source=false
checking for psol ... found
sed: -e expression #1, char 0: no previous regular expression
sed: -e expression #1, char 0: no previous regular expression
```

### PR DESCRIPTION
**Pull request by PiotrSikora**
Note: this commit doesn't add support for dynamic modules, just fixes the build.

Fixes #1110.

### CODE REVIEW

**Comment by centminmod** [comment]
thanks @PiotrSikora might need a separate patch for 1.9.32.13 too due to pagespeed 1.10 issues that still need resolving:

* https://github.com/pagespeed/ngx_pagespeed/issues/1111
* https://github.com/pagespeed/ngx_pagespeed/issues/1082

**Comment by oschaaf** [comment]
LGTM

**Comment by PiotrSikora** [comment]
@centminmod: I'll let maintainers backport it... There are some fixes in 1.10 (like interaction with [ngx_brotli](https://github.com/google/ngx_brotli)) that weren't backported to 1.9, so I'm not sure what's the process here.

File: config [hunk 1]
```diff
@@ -294,4 +294,8 @@ END
   exit 1
 fi
 
+# workaround for a bug in nginx-1.9.11, see:
+# http://hg.nginx.org/nginx/rev/ff1e625ae55b
+CFLAGS="$CFLAGS -Wno-write-strings"
```
Comment (jeffkaufman) [inline config]: Can we gate this on the nginx version?  Only change cflags if version=1.9.11 ?

**Comment by jeffkaufman** [comment]
Aside from comment about gating by version, LGTM.

I'll merge this when it's ready to go in.

**Comment by jeffkaufman** [comment]
In general, our backporting process for nginx has been that we only backport to the latest beta (1.10 in this case) because we don't have a stable release.  Following up with @centminmod on #1082 about why 1.10 isn't working for them yet.

(I'd like to start us putting out a stable release for nginx soon, in which case it would get backports for issues like this as well.)

**Comment by PiotrSikora** [comment]
Patch updated to guard use of `-Wno-write-strings`.

**Comment by jeffkaufman** [comment]
LGTM

**Comment by jeffkaufman** [comment]
merged to trunk-tracking

**Comment by jeffkaufman** [comment]
We'll need to start a new release for this change.

**Comment by jeffkaufman** [comment]
Thanks @PiotrSikora !

**Comment by rfnx** [comment]
Thanks @PiotrSikora

**Pull request merged**